collect_app/src/main/java/org/odk/collect/android/preferences/screens/AdminPreferencesFragment.java

```diff
@@ -51,6 +54,18 @@
 import static org.odk.collect.android.preferences.utilities.PreferencesUtils.displayDisabled;
 
 public class AdminPreferencesFragment extends BaseAdminPreferencesFragment implements Preference.OnPreferenceClickListener {
+
+    @Inject
+    CurrentProjectProvider currentProjectProvider;
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        DaggerUtils.getComponent(context).inject(this);
+
+        settingsProvider.getAdminSettings().save("project_name", currentProjectProvider.getCurrentProject().getName());
```

Comment:
These could use a `Preference.SummaryProvider` to just display the values (from `CurrentProjectProvider`) I think? Again, `FormMetadataPreferenceFragment` is a good example of this.

Reply:
I do this save not only to handle summary but also to display proper values in EditText when the dialog is opened since I didn't create custom dialogs and can't set that value programmatically. this is not something that I can handle using SummaryProvider right?

Reply:
I think you can use `OnBindEditTextListener` to customize the `EditText` for a setup with `SummaryProvider`. Or, I think if its `persistent` is `false` you can just do `setText` for the value you want on the Preference itself. My goal here is just to avoid writing anything to shared preferences as I'm worried that might become confusing.